### PR TITLE
doc: add document for flux

### PR DESCRIPTION
**Pull request by yangchiu**
doc: add document for flux

For https://github.com/longhorn/longhorn/issues/7335

Signed-off-by: Yang Chiu <yang.chiu@suse.com>

### CODE REVIEW

File: content/docs/1.6.0/deploy/install/install-with-flux.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
+---
+title: Install with Flux
+weight: 11
+---
+
+## Prerequisites
+- Your workstation: Install [Helm](https://helm.sh/docs/) v3.0 or later.
+- Kubernetes cluster:
+  - Ensure that each node fulfills the [installation requirements](../#installation-requirements).
+  - Follow the [official document](https://fluxcd.io/flux/installation/#install-the-flux-cli) to install Flux CLI.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/deploy/install/install-with-flux.md:10]:
```suggestion
  - [Install the Flux CLI](https://fluxcd.io/flux/installation/#install-the-flux-cli).
  - [Bootstrap Flux with GitHub](https://fluxcd.io/flux/installation/bootstrap/github/) using the Flux CLI.

    Run the following commands to export your GitHub personal access token (PAT) as an environment variable, deploy the Flux controllers on your cluster, and configure the controllers to sync the cluster state from the specified GitHub repository.

    ```bash
    export GITHUB_TOKEN=<gh-token>
    flux bootstrap github \
      --token-auth \
      --owner=<github_username> \
      --repository=<github_repo_name> \
      --branch=<branch_name> \
      --path=<folder_path_within_repo> \
      --personal
    ```
```

File: content/docs/1.6.0/deploy/install/install-with-flux.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    flux bootstrap github \
+      --token-auth \
+      --owner=<github_username> \
+      --repository=<github_repo_name> \
+      --branch=<branch_name> \
+      --path=<folder_path_within_repo> \
+      --personal
+    ```
+
+    This command initializes Flux in your cluster and sets up the sync with your specified GitOps repository.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/deploy/install/install-with-flux.md:29]: Please remove this. I moved the info to the Prerequisites section.

File: content/docs/1.6.0/deploy/install/install-with-flux.md (hunk 1 above)
Comment (jillian-maroket) [reply content/docs/1.6.0/deploy/install/install-with-flux.md:10]: I moved the bootstrap info here and expanded the code block intro to improve the procedure's overall flow.

File: content/docs/1.6.0/deploy/install/install-with-flux.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+## Installing Longhorn
+
+1. Create HelmRepository Custom Resource pointing to the Longhorn Helm chart URL:
```
Comment (jillian-maroket) [inline content/docs/1.6.0/deploy/install/install-with-flux.md:33]:
```suggestion
1. Create a HelmRepository custom resource (CR) that points to the Longhorn Helm chart URL.
```

File: content/docs/1.6.0/deploy/install/install-with-flux.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+    ```bash
+    kubectl create ns longhorn-system
+    flux create source helm longhorn-repo \
+      --url=https://charts.longhorn.io \
+      --namespace=longhorn-system \
+      --export > helmrepo.yaml
+    ```
+
+2. Create HelmRelease Custom Resource that references the HelmRepository and specifies the version of the Longhorn Helm chart to be installed:
```
Comment (jillian-maroket) [inline content/docs/1.6.0/deploy/install/install-with-flux.md:43]:
```suggestion
1. Create a HelmRelease CR that references the HelmRepository and specifies the version of the Longhorn Helm chart to be installed.
```

File: content/docs/1.6.0/deploy/install/install-with-flux.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    ```bash
+    flux create helmrelease longhorn-release \
+      --chart=longhorn \
+      --source=HelmRepository/longhorn-repo \
+      --chart-version=1.6.0 \
+      --namespace=longhorn-system \
+      --export > helmrelease.yaml
+    ```
+
+3. Verify that the GitRepo CR was created and synced successfully.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/deploy/install/install-with-flux.md:54]:
```suggestion
1. Verify that the HelmRelease CR was created and synced successfully.
```

File: content/docs/1.6.0/deploy/install/install-with-flux.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+    ```bash
+    flux get helmrelease longhorn-release -n longhorn-system
+    ```
+
+4. Verify that Longhorn was installed successfully.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/deploy/install/install-with-flux.md:60]:
```suggestion
1. Verify that Longhorn was installed successfully.
```

File: content/docs/1.6.0/deploy/install/install-with-flux.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+    csi-resizer-6d8cf5f99f-68cw7                        1/1     Running   0          101s
+    csi-attacher-7bf4b7f996-df8v6                       1/1     Running   0          101s
+    csi-attacher-7bf4b7f996-g9cwc                       1/1     Running   0          101s
+    csi-attacher-7bf4b7f996-8l9sw                       1/1     Running   0          101s
+    csi-resizer-6d8cf5f99f-smdjw                        1/1     Running   0          101s
+    instance-manager-b34d5db1fe1e2d52bcfb308be3166cfc   1/1     Running   0          114s
+    engine-image-ei-df38d2e5-cv6nc                      1/1     Running   0          114s
+    ```
+
+5. [Create an NGINX Ingress controller with basic authentication](../../accessing-the-ui/longhorn-ingress) to access the Longhorn UI. Authentication to the Longhorn UI is not enabled by default.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/deploy/install/install-with-flux.md:90]:
```suggestion
1. [Create an NGINX Ingress controller with basic authentication](../../accessing-the-ui/longhorn-ingress) to access the Longhorn UI. Authentication to the Longhorn UI is not enabled by default.
```

File: content/docs/1.6.0/deploy/install/install-with-flux.md [hunk 8]
```diff
@@ -0,0 +1,103 @@
...
+
+6. [Access the Longhorn UI](../../accessing-the-ui).
```
Comment (jillian-maroket) [inline content/docs/1.6.0/deploy/install/install-with-flux.md:92]:
```suggestion
1. [Access the Longhorn UI](../../accessing-the-ui).
```

File: content/docs/1.6.0/deploy/install/install-with-flux.md [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+
+## Continuous Operations via GitOps
+
+1. Commit and push exported manifests to your GitOps Repository:
```
Comment (jillian-maroket) [inline content/docs/1.6.0/deploy/install/install-with-flux.md:96]:
```suggestion
You can commit and push exported manifests to your GitOps repository.
```

File: content/docs/1.6.0/deploy/install/install-with-flux.md [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+
+    ```bash
+    git add helmrepo.yaml helmrelease.yaml
+    git commit -m "Add HelmRepository and HelmRelease for Longhorn installation"
+    git push origin <branch_name>
+   ```
+   From this point forward, any modifications or updates to the HelmRelease or HelmRepository can be made by editing the YAML manifests in your GitOps repository. Flux will automatically detect these changes and apply them to your Kubernetes cluster accordingly, enabling further operations via Git push without direct access to the Kubernetes cluster.
```
Comment (jillian-maroket) [inline content/docs/1.6.0/deploy/install/install-with-flux.md:103]:
```suggestion
Afterwards, you can modify the HelmRelease and HelmRepository CRs by editing the YAML manifests in your GitOps repository. Flux automatically detects and applies the changes without requiring direct access to your Kubernetes cluster.
```

File: content/docs/1.6.0/deploy/install/install-with-flux.md (hunk 5 above)
Comment (jillian-maroket) [reply content/docs/1.6.0/deploy/install/install-with-flux.md:54]: The best practice is to use "1." for all ordered list items.

**Review by jillian-maroket — COMMENTED** [review]
@yangchiu Let me know if you have concerns about the changes.

**Comment by yangchiu** [comment]
Refined based on @jillian-maroket 's suggestions. Thanks!

**Review by jillian-maroket — APPROVED** [review]

**Review by innobead — APPROVED** [review]

**Pull request merged**